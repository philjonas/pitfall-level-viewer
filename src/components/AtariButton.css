.atari-button{
    cursor: pointer;
    border: 1px dashed lightgreen;
    color: lightgreen;
    padding: 4px;
    user-select: none;
}