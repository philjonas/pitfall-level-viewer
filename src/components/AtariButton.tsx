import React from 'react';
import './AtariButton.css';

export interface AtariButtonProps {
  label?: string;
  onClick?: () => void;
}

export const AtariButton = ({
  label,
  onClick = () => {},
}: AtariButtonProps): JSX.Element => {
  return (
    <div onClick={onClick} data-testid="btn" className={'atari-button'}>
      <span data-testid="txt">{label}</span>
    </div>
  );
};
