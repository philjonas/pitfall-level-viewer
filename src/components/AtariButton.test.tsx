import {render, fireEvent} from '@testing-library/react';
import {AtariButtonProps, AtariButton} from './AtariButton';

const setup = (props?: AtariButtonProps) => {
  const utils = render(<AtariButton {...props} />);
  const btn = utils.getByTestId('btn');
  const label = utils.getByTestId('txt');
  return {
    btn,
    label,
    ...utils,
  };
};

describe('AtariButton', () => {
  it('renders correctly', () => {
    const wrapper = <AtariButton label="button" />;
    expect(wrapper).toMatchSnapshot();
  });

  it('returns the correct label', () => {
    const labelTXT = 'Right';
    const {label} = setup({label: labelTXT});
    expect(label.innerHTML).toBe(labelTXT);
  });

  it('fires when clicked', () => {
    const props: AtariButtonProps = {label: 'OK', onClick: jest.fn()};
    const {btn} = setup(props);
    expect(props.onClick).toBeCalledTimes(0);
    fireEvent.click(btn);
    expect(props.onClick).toBeCalledTimes(1);
  });
});
