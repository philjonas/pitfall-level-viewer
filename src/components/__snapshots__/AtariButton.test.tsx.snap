// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`AtariButton renders correctly 1`] = `
<AtariButton
  label="button"
/>
`;
